src/gen/CMakeLists.txt

```diff
@@ -15,4 +15,5 @@ endif()
 
 target_link_libraries(gen
   base-lib
-  ${TINYXML_LIBRARY})
+  ${TINYXML_LIBRARY}
+  ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
I think we should add `target_link_libraries(base-lib ${CMAKE_THREAD_LIBS_INIT})` in the `base/CMakeLists.txt` instead of adding it here. (So any application linked with base-lib will be linked with threads libs.)